src/modules/Comment/controller.ts

```diff
@@ -45,4 +45,37 @@ const commentOnNewsArticle = async (req: Request, res: Response) => {
   }
 };
 
-export default { commentOnNewsArticle };
+const getNewsComments = async (req: Request, res: Response) => {
+  const { url } = req.body;
```

Comment:
Since this is a get request, the url should come from `req.query`

Reply:
The Typescript Error:
```
Type 'string | string[] | ParsedQs | ParsedQs[] | undefined' is not assignable to type 'string | FindOperator<string> | undefined'.
  Type 'string[]' is not assignable to type 'string | FindOperator<string> | undefined'.ts(2322)
(property) url?: string | FindOperator<string> | undefined
```
says that the `url` can be `'string | string[] | ParsedQs | ParsedQs[] | undefined'`. So by checking if it isn't a string do a return, typescript will understand that it is indeed a string.

[Argument of type 'string | ParsedQs | string[] | ParsedQs[] | undefined' is not assignable to parameter of type 'string'.ts(2345) [Github link in comments]](https://www.reddit.com/r/typescript/comments/tv3gci/comment/i378ohg/)

Reply:
When did you get this error ? 
i dont have it on my end

Reply:
When I implemented the `GET` and `req.query`, this error showed on the `findOneBy({ url })`
```
 const { url } = req.query;
  try {
    const news = await newsRepository.findOneBy({ url })
    ...
```
but it was resolved with this:
```
const { url } = req.query;

  if (typeof url !== 'string') {
    res.status(400).send({ message: 'Invalid url.' });
    return;
  }

  try {
    const news = await newsRepository.findOneBy({ url })
    ...
```

and the `router.get` worked, so I could get all the comments from one news article with `GET` request